### PR TITLE
[Admin] find Post with content

### PR DESCRIPTION
**Pull request by ghost**
## Description
Find Post with input content
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- app/Providers/RouteServiceProvider.php
- routes/web.php

## SQL query and `explain`

`explain select count(*) as aggregate from `posts` where `content` like '%lorem%'\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: posts
>    partitions: NULL
>          type: ALL
> possible_keys: NULL
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 16
>      filtered: 11.11
>         Extra: Using where
> 1 row in set, 1 warning (0.00 sec)

`explain select * from `posts` where `content` like '%lorem%' limit 15 offset 0\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: posts
>    partitions: NULL
>          type: ALL
> possible_keys: NULL
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 16
>      filtered: 11.11
>         Extra: Using where
> 1 row in set, 1 warning (0.00 sec)

`explain select * from `users` where `users`.`id` in ('1', '4', '5', '6', '7', '8', '9', '10')\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: users
>    partitions: NULL
>          type: ALL
> possible_keys: PRIMARY
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 10
>      filtered: 80.00
>         Extra: Using where
> 1 row in set, 1 warning (0.01 sec)

`explain select * from `products` where `products`.`id` in ('1', '3', '4', '6', '8', '9', '10', '12', '18', '19')\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: SIMPLE
>         table: products
>    partitions: NULL
>          type: ALL
> possible_keys: PRIMARY
>           key: NULL
>       key_len: NULL
>           ref: NULL
>          rows: 24
>      filtered: 41.67
>         Extra: Using where
> 1 row in set, 1 warning (0.00 sec)

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -21,6 +21,7 @@
     Route::resource('categories', 'CategoryController')->parameters(['categories' => 'id']);
     Route::resource('products', 'ProductController')->parameters(['products' => 'id']);
     Route::resource('posts', 'PostController')->parameters(['posts' => 'id']);
+    Route::post('posts/q', 'PostController@findByContent')->name('posts.find');
```
Comment (ducxinh-dn-201) [inline routes/web.php]: k cần thêm `route mới đâu em` dùng chung với cái posts.index ấy.

File: resources/views/admin/pages/posts/index.blade.php [hunk 2]
```diff
@@ -9,13 +9,26 @@
...
-                <button class="btn btn-default" type="button">{{ __('post.admin.list.go') }}</button>
-              </span>
-            </div>
+          <div class="col-md-10 col-sm-10 col-xs-12 form-group pull-right top_search">
+            <form action="{{ route('admin.posts.find') }}" method="POST">
+              @csrf
+              <div class="input-group">   
+                <div class="col-md-6">
+                  <select name="post_status" class="form-control">
+                    <option value="-1">{{ __('post.admin.list.select_title') }}</option>
```
Comment (ducxinh-dn-201) [inline resources/views/admin/pages/posts/index.blade.php]: `value` em nên dùng `const`

File: routes/web.php (hunk 1 above)
Comment (vunamphuong17) [reply routes/web.php]: em add để post dữ liệu lên ạ chứ post lên chỗ `index` sẽ trùng với cái `add posts` rồi ạ :3 em thêm `route` thôi chứ vẫn dùng page index ạ.

File: resources/views/admin/pages/posts/index.blade.php (hunk 2 above)
Comment (vunamphuong17) [reply resources/views/admin/pages/posts/index.blade.php]: dùng const của `model` luôn cx được hả anh.

File: app/Http/Controllers/Admin/PostController.php [hunk 3]
```diff
@@ -11,16 +11,45 @@ class PostController extends Controller
...
      * Display a listing of the resource.
      *
+     * @param \Illuminate\Http\Request $request request content
+     *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
+        if ($request->content || $request->post_status == 0 || $request->post_status == 1) {
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Admin/PostController.php]: $request->post_status == 0 thì case request k có param `post_status` cũng pass đấy em

Comment (at-xinhnguyen) [reply app/Http/Controllers/Admin/PostController.php]: chỗ này chỉ cẩn isset($request->post_status) || isset($request->content)

File: resources/views/admin/pages/posts/index.blade.php [hunk 4]
```diff
@@ -9,13 +9,25 @@
...
-              <span class="input-group-btn">
-                <button class="btn btn-default" type="button">{{ __('post.admin.list.go') }}</button>
-              </span>
-            </div>
+          <div class="col-md-10 col-sm-10 col-xs-12 form-group pull-right top_search">
+            <form action="{{ route('admin.posts.index') }}" method="GET">
+              <div class="input-group">   
+                <div class="col-md-6">
+                  <select name="post_status" class="form-control">
+                    <option value="{{ App\Models\Post::STATUS_DEFAULT }}">{{ __('post.admin.list.select_title') }}</option>
```
Comment (at-xinhnguyen) [inline resources/views/admin/pages/posts/index.blade.php]: option k lựa chọn thì em nên để trống để dex handle

File: app/Models/Post.php [hunk 5]
```diff
@@ -7,8 +7,12 @@
 class Post extends Model
 {
     protected $table = 'posts';
+    
     const TYPE_REVIEW = 1;
     const TYPE_COMMENT = 2;
+    const APPROVED = 1;
+    const UNAPPROVED = 0;
+    const STATUS_DEFAULT = -1;
```
Comment (at-xinhnguyen) [inline app/Models/Post.php]: k cần thiết const này đâu em

File: app/Http/Controllers/Admin/PostController.php (hunk 3 above)
Comment (at-xinhnguyen) [reply app/Http/Controllers/Admin/PostController.php]: k cần viết hàm `findByContent ` đâu em. chỉ mỗi hàm index đc rồi
      $posts = Post::with(['user', 'product'])
        ->when(isset(request()->content), function ($query) {
            return $query->where('content', 'like', '%' . request()->content . '%');
        })
        ->when(isset(request()->post_status), function ($query) {
            return $query->where('status', (int) request()->post_status);
        })->paginate($perPage);

File: resources/views/admin/pages/posts/index.blade.php [hunk 6]
```diff
@@ -9,13 +9,24 @@
...
-              <span class="input-group-btn">
-                <button class="btn btn-default" type="button">{{ __('post.admin.list.go') }}</button>
-              </span>
-            </div>
+          <div class="col-md-10 col-sm-10 col-xs-12 form-group pull-right top_search">
+            <form action="{{ route('admin.posts.index') }}" method="GET">
+              <div class="input-group">   
+                <div class="col-md-6">
+                  <select name="post_status" class="form-control">
+                      <option value="">{{ __('post.admin.list.subtitle_index') }}</option>
```
Comment (at-xinhnguyen) [inline resources/views/admin/pages/posts/index.blade.php]: fix convention em. remove `2 space`

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**